torch/_inductor/cudagraph_trees.py

```diff
@@ -1972,6 +1994,23 @@ def _run(self, new_inputs: List[Tensor], function_id: FunctionID):
             if self.current_node is not None:
                 self.apply_checkpoint_execution_state_in_allocator()
 
+            # re-record due to static input tensor address changes
```

Comment:
If we have 

node1 ---> node2

And we hit the limit on node2, then we'll be applying the checkpointing state on the allocator in each iteration. applying the checkpointing state in the allocator is slow. We checkpoint the allocator state because we have tensors in the cudagraph caching allocator pool which are unallocated, and we want to do more allocations to it. Here, we're about to do more allocations in the eager pool so we dont need to checkpoint it.